src/main/java/org/testng/internal/Parameters.java

```diff
@@ -102,10 +102,14 @@
 
     mapping.put(BeforeMethod.class.getSimpleName(), beforeAfterMethod);
     mapping.put(AfterMethod.class.getSimpleName(), beforeAfterMethod);
-    mapping.put(Test.class.getSimpleName(), Arrays.asList(new Class<?>[] {ITestContext.class}));
+    mapping.put(Test.class.getSimpleName(), newList(ITestContext.class));
 
   }
 
+  private static List<Class<?>> newList(Class<?>...elements) {
```

Comment:
This won't work because the `Arrays.asList(ITestContext.class, XmlTest.class);` when invoked results in a `List<Class<? extends Serializable>` (both `ITestResult` and `XmlTest` implement `Serializable` and so Java deduces this type. I tried this already and since it wasn't working is why I resorted to this approach.